src/lib/process/api.ts

```diff
@@ -0,0 +1,97 @@
+import { Restrictions } from '../types/restrictions.js'
+import { Options } from '../types/options.js'
+import { Polkadot } from '../types/substrate.js'
+import { Process } from './index.js'
+
+export const createProcessTransaction = async (
+  polkadot: Polkadot,
+  processId: string,
+  restrictions: Restrictions,
+  options: Options
+): Promise<Process> => {
+  const sudo = polkadot.keyring.addFromUri(options.USER_URI)
+
+  return new Promise((resolve, reject) => {
+    let unsub: Function
+    polkadot.api.tx.sudo
+      .sudo(polkadot.api.tx.processValidation.createProcess(processId, restrictions))
+      .signAndSend(sudo, (result: any) => {
+        if (result.status.isInBlock) {
+          const { event } = result.events.find(
+            ({ event: { method } }: { event: { method: string } }) => method === 'ProcessCreated'
+          )
+
+          const data = event.data
+          const newProcess: Process = {
+            id: processId,
+            version: data[1].toNumber(),
+            status: 'Enabled',
+            restrictions: restrictions,
+          }
+
+          unsub()
+          resolve(newProcess)
+        }
+      })
+      .then((res: Function) => {
+        unsub = res
+      })
+      .catch((err: Error) => {
+        reject(err)
+      })
+  })
+}
+
+export const disableProcessTransaction = async (
+  polkadot: Polkadot,
```

Comment:
Since payloads and responses are the same for create and disable we can abstract and update 

```ts
type ProcessTransactionFn = (polkadot: Polkadot) => Promise<Process>
```

Or in some other way, but anyway your implementation covers everything.

Also might be a good idea to keep types separate like in `/types` folder or declare under namescape and point via config so you can use like:


```ts
const aFn: Process.AFn = (arg1, arg2) => Promise.resolve(Process)
```

I can already see you are importing Process and etc. there is config prop to do the auto import, also might need to be wrapped in namescape declaration

Reply:
More on this here -> https://www.typescriptlang.org/tsconfig

```
[#](https://www.typescriptlang.org/tsconfig#typeRoots)
Type Roots -
typeRoots
By default all visible ”@types” packages are included in your compilation. Packages in node_modules/@types of any enclosing folder are considered visible. For example, that means packages within ./node_modules/@types/, ../node_modules/@types/, ../../node_modules/@types/, and so on.

If typeRoots is specified, only packages under typeRoots will be included. For example:

{
  "[compilerOptions](https://www.typescriptlang.org/tsconfig#compilerOptions)": {
    "[typeRoots](https://www.typescriptlang.org/tsconfig#typeRoots)": ["./typings", "./vendor/types"]
  }
```

Reply:
then restrictyions could be optional with the question mark '?'

Reply:
Use the `parameter?: <type>` syntax to make a parameter optional.